packages/space-opera/src/components/camera_settings/reducer.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import {registerStateMutator, State} from '../../redux/space_opera_base.js';
+import {SphericalPositionDeg, Vector3D} from '../../redux/state_types.js';
+
+/*
+ * Register state mutators and get corresponding dispatchers.
+ */
+
+const SET_CAMERA_CONTROLS_ENABLED = 'SET_CAMERA_CONTROLS_ENABLED';
```

Comment:
I feel like these lines aren't adding much value. I'd either put them back how they were, or export a union type so TS can check these for errors. You can take a look at `ARStatus` in ARRenderer.ts for an example.

Reply:
this is me setting up for an actual reducer

Reply:
these are usually exported so they can be used in multiple places, at the moment, I am just moving everything into the respective location

Reply:
In that case, I'd recommend going with the `ARStatus` approach so you can also get type checking. But you can do that down the road in your refactor.